bot/utils/__init__.py

```diff
@@ -66,6 +70,34 @@ def _convert_keys(self) -> None:
             self.__setitem__(k, v)
 
 
+class ProxyUser(Object):
+    """
+    Create a proxy user object from the given id.
+
+    Used when a Member or User object cannot be resolved.
+    """
+
+    def __init__(self, user_id: str):
+        log.trace(f"Attempting to create a proxy user for the user id {user_id}.")
+        try:
+            user_id = int(user_id)
+        except ValueError:
+            raise BadArgument
+
+        super().__init__(user_id)
+        self.bot = False
+        self.display_name = str(user_id)
+        self.mention = f"<@{self.id}>"
+
+    @staticmethod
+    def avatar_url_as(*_, **__) -> None:
+        """Return `None` when avatar is requested."""
+        return None
+
+    def __str__(self):
+        return f"User: ID {self.id}"
```

Comment:
Is this actually used anywhere?

Reply:
The str shows up in the infraction user search

Reply:
I don't see how it gets used there.